code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	colour = "cerulean"
+	runepath = /obj/effect/warped_rune/ceruleanspace
+
+/obj/effect/warped_rune/ceruleanspace
+	desc = "A shadow of what once passed these halls, a memory perhaps?"
+	icon_state = "holo_rune"
+	var/obj/effect/overlay/holotile
+
+
+/obj/effect/warped_rune/ceruleanspace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/holoform)
+
+///makes a hologram of the mob stepping on the tile, any new person stepping in will replace it with a new hologram
+/obj/effect/warped_rune/ceruleanspace/proc/holoform()
+
+	for(var/mob/living/L in T)
+		if(locate(holotile) in T)//here to avoid multiple person on the rune related bugs
+			qdel(holotile)
+
+		holotile = new(T) //setting up the hologram to look like the person that just stepped in
+		holotile.icon = L.icon
+		holotile.icon_state = L.icon_state
+		holotile.alpha = 100
+		holotile.name = "[L.name] (Hologram)"
+		holotile.add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
+		holotile.copy_overlays(L, TRUE)
+
+///destroys the hologram with the rune
+/obj/effect/warped_rune/ceruleanspace/Destroy()
+	..() //needed so the rune still gets stored by the crossbreed
+	qdel(holotile)
+
+
+
+
+
+/obj/item/slimecross/warping/pyrite
+	colour = "pyrite"
+	runepath = /obj/effect/warped_rune/pyritespace
+	effect_desc = "draws a rune that will randomly color whoever steps on it"
+
+/obj/effect/warped_rune/pyritespace
+	desc = "Who shall we be today? they asked, but not even the canvas would answer."
+	icon_state = "colorune"
+
+/obj/effect/warped_rune/pyritespace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/colored)
+
+///colors whoever steps on the rune randomly
+/obj/effect/warped_rune/pyritespace/proc/colored()
+	for(var/mob/living/L in T)
+		L.color = rgb(rand(0,255),rand(0,255),rand(0,255))
+
+
+/* Will make anyone on the rune do much MUCH more punch damage. Doesn't boost normal melee weapon damage*/
+
+/obj/item/slimecross/warping/red
+	colour = "red"
+	runepath = /obj/effect/warped_rune/redspace
+	effect_desc = "draw a rune severely increasing your punch damage. Only works on one person at a time."
+    drawing_time = 100 //longer to draw so you don't draw it in the middle of a fight.
+
+/obj/effect/warped_rune/redspace
+	desc = "Progress is made through adversity, power is obtained through violence"
+	icon_state = "rage_rune"
+	runepath = /obj/effect/warped_rune/redspace
+	var/mob/living/carbon/human/H
+	var/datum/species/og_punch_low //used to remember the original low punch force of H
+	var/og_punch_high //used to remember the original high punch force of H
+	var/rage = FALSE //here to check if someone is already enraged on the rune
+	var/og_threshold
+
+/obj/effect/warped_rune/redspace/Initialize()
+	.=..()
+	RegisterSignal(T,COMSIG_ATOM_ENTERED,.proc/angry_in)
+	RegisterSignal(T,COMSIG_ATOM_EXITED,.proc/angry_out)
+
+
+/obj/effect/warped_rune/redspace/proc/angry_in()
+
+	if(rage) //don't buff anyone if someone is already buffed by the rune.
```

Comment:
shouldn't this be inside of the for loop?

honestly, why not just make this work on multiple people at the same time? you can move some of this stuff into a status effect so that the original punch damage thresholds don't have to be stored in this rune. or you can just not bother fiddling with punch damage at all and instead use a martial art of some kind.

proposal for a new effect for this rune: teaches anyone who crosses it the psychotic brawling martial art (and colors them red) until they're doused with water.

proposal for a much easier to implement alternative new effect for this rune: adds bath salts to the system of anyone who crosses it (or who stays on it).

Reply:
The only reason I didn't make it work on multiple people at the same time is so that I could redistribute the right punch stats to the person that is leaving instead of the stats of someone else. Although I probably could've worked around that with some list fuckery